src/device/mod.rs

```diff
@@ -1422,9 +1440,23 @@ impl Runtime {
             return Err(Error::EndpointNotProvided);
         }
 
-        self.requested_state.exit_node = Some(exit_node);
+        let old_exit_node = self.requested_state.exit_node.replace(exit_node);
         wg_controller::consolidate_wg_state(&self.requested_state, &self.entities, &self.features)
-            .await
+            .await?;
+
+        let old_exit_pubkey = old_exit_node
```

Comment:
I think in this situation we will be killing connections not just on reconnect, but also on disconnect.

E.g. if `exit_node` is None, but `self.requested_state.last_exit_node` is `Some(...)` then we still call `.reset_existing_connections(...)`. In super practical terms it would mean, that on android the killing will happen twice, once per OS mechanism, and once in ours.

But IMHO, this may be ok todo for "extra safety". I will request Dragan's opinion too :D

Reply:
I'm curious in which cases should `exit_node` be None here?

On disconnect we currently stop telio, so it should retain `last_exit_node`, if I got it right.
But even if we change this eventually, it shouldn't be much of an overhead.

Reply:
Ok, so I will wait for the decision